src/presenced.cpp

```diff
@@ -281,10 +291,18 @@ Client::reconnect(const std::string& url)
                 return promise::Error("No valid URL provided and current URL is not valid");
         }
 
-        setConnState(kConnecting);
-        return retry("presenced", [this](int no)
+        auto wptr = weakHandle();
+        if (mRetryCtrl)
         {
+            mRetryCtrl->abort();
+        }
+        mRetryCtrl.reset(createRetryController("presenced", [wptr, this](int no) -> Promise<void>
+        {
+            if (wptr.deleted())
+                return promise::Error("destroyed");
```

Comment:
In line 302, where the `presence::Client` has been deleted, the code returns a promise. However, I've seen how heartbeats / retry controller results in new connection attempts even when the Client is already defunct.
With my comment, I'm just calling your attention to that point.